packages/rum/test/specs/apm-base.spec.js

```diff
@@ -34,41 +35,32 @@ var enabled = bootstrap()
 const { serviceName, serverUrl } = getGlobalConfig('rum').agentConfig
 
 describe('ApmBase', function() {
-  var serviceFactory
+  let serviceFactory
+  let apmBase
   beforeEach(function() {
     serviceFactory = createServiceFactory()
+    apmBase = new ApmBase(serviceFactory, !enabled)
   })
 
-  it('should send page load metrics before or after load event', function(done) {
-    var apmBase = new ApmBase(serviceFactory, !enabled)
-    var trService = serviceFactory.getService('TransactionService')
+  it('should send page load metrics after load event', done => {
     apmBase.config({ serviceName, serverUrl })
     apmBase._sendPageLoadMetrics()
-    var tr = trService.getCurrentTransaction()
+    var tr = apmBase.getCurrentTransaction()
     expect(tr.name).toBe('Unknown')
     expect(tr.type).toBe(PAGE_LOAD)
     spyOn(tr, 'detectFinish').and.callThrough()
-    window.addEventListener('load', function() {
-      setTimeout(() => {
-        expect(tr.detectFinish).toHaveBeenCalled()
-        apmBase.setInitialPageLoadName('new page load')
-        expect(document.readyState).toBe('complete')
-
-        apmBase._sendPageLoadMetrics()
-        tr = trService.getCurrentTransaction()
-        expect(tr.name).toBe('new page load')
-        expect(tr.type).toBe(PAGE_LOAD)
-        spyOn(tr, 'detectFinish')
-        setTimeout(() => {
-          expect(tr.detectFinish).toHaveBeenCalled()
-          done()
-        })
-      })
+
+    apmBase.setInitialPageLoadName('new page load')
+    apmBase.observe(TRANSACTION_END, endedTr => {
+      expect(document.readyState).toBe('complete')
+      expect(tr.detectFinish).toHaveBeenCalled()
+      expect(endedTr.name).toBe('new page load')
+      expect(endedTr.type).toBe(PAGE_LOAD)
+      done()
     })
   })
 
   it('should disable all auto instrumentations when instrument is false', () => {
-    const apmBase = new ApmBase(serviceFactory, !enabled)
```

Comment:
Not strictly related to this PR but I remember that we had a test `new ApmBase(serviceFactory, true)` that tests the agent when it's disabled in the environment. I can't find it now, do you remember why we removed that?

Reply:
It was on apm-rum-react if i remember correctly.